workspaces/diff-engine/src/events/spec_chunk.rs

```diff
@@ -0,0 +1,332 @@
+        {"ContributionAdded": {"id": "path_1.GET","key": "purpose","value": "todos"}}
+      ]),
+    ),
+    (
+      String::from("0002.json"),
+      false,
+      // Adding a simple shape to the GET /todos response
+      json!([
+        {"BatchCommitStarted": {"batchId": "batch-2", "parentId": "batch-1", "commitMessage": "dsasa" }},
+        {"ShapeAdded": { "shapeId": "object_shape_1", "baseShapeId": "$object", "parameters": { "DynamicParameterList": { "shapeParameterIds": [] } }, "name": "" }},
+        {"ShapeAdded": {"shapeId": "boolean_shape_1","baseShapeId": "$boolean","parameters": {"DynamicParameterList": {"shapeParameterIds": []}},"name": ""}},
+        {"FieldAdded": {"fieldId": "field_1","shapeId": "object_shape_1","name": "isDone","shapeDescriptor": {"FieldShapeFromShape": {"fieldId": "field_1","shapeId": "boolean_shape_1"}}}},
+        {"ResponseBodySet": {"responseId": "response_1","bodyDescriptor": {"httpContentType": "application/json","shapeId": "object_shape_1","isRemoved": false}}},
+        {"BatchCommitEnded": { "batchId": "batch-2" }}
+      ]),
+    ),
+    (
+      String::from("0001.json"),
+      false,
+      // Adding a Request and Response for GET /todos
+      json!([
+        {"BatchCommitStarted": {"batchId": "batch-1", "parentId": "root", "commitMessage": "Add Request and Response for GET /todos" }},
+        {"RequestAdded": { "requestId": "request_1","pathId": "path_1","httpMethod": "GET" }},
+        {"ResponseAddedByPathAndMethod": { "responseId": "response_1", "pathId": "path_1", "httpMethod": "GET", "httpStatusCode": 200 }},
+        {"BatchCommitEnded": { "batchId": "batch-1" }}
+      ]),
+    ),
+  ].into_iter().map(|(file_name, is_root, events_json)| {
+    let events : Vec<SpecEvent> = serde_json::from_value(events_json).expect("example events should be valid spec events");
+
+    SpecChunkEvent::from((file_name, is_root, events))
+  }).collect();
+
+    dbg!(&spec_chunk_events);
+
+    assert_eq!(
+      spec_chunk_events
+        .iter()
+        .map(|chunk| matches!(chunk, SpecChunkEvent::Root(_)))
+        .collect::<Vec<_>>(),
+      vec![true, false, false]
+    );
+
+    assert_eq!(
+      spec_chunk_events
+        .iter()
+        .map(|chunk| matches!(chunk, SpecChunkEvent::Batch(_)))
+        .collect::<Vec<_>>(),
+      vec![false, true, true]
+    );
+  }
+
+  #[test]
+  pub fn constructs_valid_batch_chunks() {
+    let valid_batch_events = serde_json::from_value::<Vec<SpecEvent>>(json!([
+      {"BatchCommitStarted": {"batchId": "batch-1", "parentId": "root", "commitMessage": "Add Request and Response for GET /todos" }},
+      {"RequestAdded": { "requestId": "request_1","pathId": "path_1","httpMethod": "GET" }},
+      {"ResponseAddedByPathAndMethod": { "responseId": "response_1", "pathId": "path_1", "httpMethod": "GET", "httpStatusCode": 200 }},
+      {"BatchCommitEnded": { "batchId": "batch-1" }}
+    ])).unwrap();
```

Comment:
Here are tests covering all of the `Batch` parsing logic, using snapshots to verify each case produces the error we expect it to.